pkg/component/runtime/manager_test.go

```diff
@@ -3528,20 +3538,19 @@ LOOP:
 	subCancel()
 	cancel()
 
-	// check progression, require stop fake-0 before start fake-1
 	stateProgressionWG.Wait()
-	comp0Stopped := false
-	for _, step := range stateProgression {
-		if step.componentID == IDComp0 &&
-			step.state.State == client.UnitStateStopped {
-			comp0Stopped = true
-		}
-		if step.componentID == IDComp1 &&
-			step.state.State == client.UnitStateStarting {
-			require.True(t, comp0Stopped)
-			break
-		}
-	}
+
+	// check progressions:
+	// require stop fake-0 Starting -> Healthy -> Stopped
+	require.Truef(t, len(comp0Progression) >= 3, "comp0Progression should have at least 3 state changes. comp0Progression: %v", comp0Progression)
+	require.Equalf(t, client.UnitStateStarting, comp0Progression[0].state.State, "Comp0 must start with state Starting comp0Progression: %v", comp0Progression)
```

Comment:
Minor grammar nit: there should be a period there: `Comp0 must start with state Starting. comp0Progression:`